packages/lesswrong/server/resolvers/ultraFeedResolver.ts

```diff
@@ -169,118 +138,97 @@ const createSourcesMap = (
   sourceWeights: Record<string, number>,
   postThreadsItems: FeedFullPost[],
   commentThreadsItems: FeedCommentsThread[],
-  spotlightItems: FeedSpotlight[]
-): Record<FeedItemSourceType, WeightedSource> => {
-  const sources = {} as Record<FeedItemSourceType, WeightedSource>;
-  const commentSourceType: FeedItemSourceType = 'recentComments';
-  
-  Object.entries(sourceWeights).forEach(([source, weight]) => {
-    const sourceType = source as FeedItemSourceType;
-    if (weight <= 0) return;
-
-    let renderAsType: FeedItemRenderType;
-    if ((feedPostSourceTypesArray as readonly string[]).includes(sourceType)) {
-      renderAsType = 'feedPost';
-    } else if (sourceType === commentSourceType) {
-      renderAsType = 'feedCommentThread';
-    } else if ((feedSpotlightSourceTypesArray as readonly string[]).includes(sourceType)) {
-      renderAsType = 'feedSpotlight';
-    } else {
-      // eslint-disable-next-line no-console
-      console.warn(`UltraFeedResolver: Source type "${sourceType}" found in sourceWeights but not in known type arrays.`);
-      return; // Skip sources not mappable to a render type
-    }
-
-    sources[sourceType] = {
-      weight,
-      items: [], 
-      renderAsType
-    };
+  spotlightItems: FeedSpotlight[],
+  bookmarkItems: PreparedBookmarkItem[]
+): Partial<Record<FeedItemSourceType, WeightedSource>> => {
+  const sources: Partial<Record<FeedItemSourceType, WeightedSource>> = {};
+  Object.entries(sourceWeights).forEach(([src, w]) => {
+    if (w > 0) sources[src as FeedItemSourceType] = { weight: w, items: [] };
```

Comment:
this is too dense to be on one line.

Also, what happened to the rest of the code?

Reply:
The refactor here is that instead of assigning "renderTypes" to sources, we assign them to individual items (because bookmarks are a single source but individual items can have different render types). Changing this actually took out a lot of code